src/main/java/com/hotels/beeju/extensions/BeejuJUnitExtension.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.extensions;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.thrift.TException;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hotels.beeju.core.BeejuCore;
+
+abstract public class BeejuJUnitExtension implements BeforeEachCallback, AfterEachCallback {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BeejuJUnitExtension.class);
+  public BeejuCore core;
+  private Path tempDirectory;
+
+  public BeejuJUnitExtension(String databaseName, Map<String, String> configuration) {
+    core = new BeejuCore(databaseName, configuration);
+  }
+
+  /**
+   * Initialise the warehouse path.
+   * <p>
+   * This method can be overridden to provide additional initialisations.
+   * </p>
+   *
+   * @throws Throwable If the initialisation fails.
+   */
+  protected void init() throws Throwable {
+    tempDirectory = Files.createTempDirectory("root");
+    core.setHiveVar(HiveConf.ConfVars.METASTOREWAREHOUSE, tempDirectory.toString());
+  }
+
+  /**
+   * Implement this method to prepare the rule for a new test.
+   * <p>
+   * This is called after the rule is initialised.
+   * </p>
+   *
+   * @throws Throwable If the rule cannot be prepared for a new test.
+   */
+  protected abstract void beforeTest() throws Throwable;
+
+  @Override
+  public void beforeEach(ExtensionContext context) throws TException {
+    try {
+      init();
+      beforeTest();
+    } catch (Throwable throwable) {
+      LOG.error("Unable to initialise BeejuJUnit extension", throwable);
```

Comment:
Not really useful, imagine running a test that is not working and you only have this this error in the logs somewhere., that's hard to debug. Just remove the try catch and let all the exceptions bubble up. Fail fast